test/validation/shared.go

```diff
@@ -142,13 +142,14 @@ func ValidateRPResources(ctx context.Context, t *testing.T, expected *RPResource
 			// Validate expected output resources are present in the response
 			if len(expectedResource.OutputResources) > 0 {
 				t.Log("validating output resources")
-				bytes, err := json.Marshal(res.Properties["status"])
+				status := res.Properties["status"].(map[string]interface{})
+				or := status["outputResources"].([]interface{})
+				bytes, err := json.Marshal(or)
 				require.NoError(t, err)
 
-				var outputResourcesMap map[string][]OutputResourceResponse
-				err = json.Unmarshal(bytes, &outputResourcesMap)
+				var outputResources []OutputResourceResponse
+				err = json.Unmarshal(bytes, &outputResources)
```

Comment:
Can you tell me a little bit more about what's going on here? It looks like `outputResources` might be either an array or map. That doesn't sound good if that's the case.

Reply:
Earlier, the only fields in status were "outputResources" and therefore status looked like map with key = "outputresources" and value = resources and we were directly calling marshal on the status. Now, we have added recipe.
![Screenshot 2023-10-30 at 4 06 50 PM](https://github.com/radius-project/radius/assets/28875764/af5d97ac-7016-4df9-bddc-b7f7c59fbba7)

Therefore, we are now separating out status["outputResources"] and then marshaling and unmarshaling before validating the output resources.